op-node/rollup/derive/attributes.go

```diff
@@ -118,3 +134,43 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 		GasLimit:              (*eth.Uint64Quantity)(&sysConfig.GasLimit),
 	}, nil
 }
+
+var bscDefaultGasPrice = big.NewInt(5000000000)
+
+func (ba *FetchingAttributesBuilder) prepareAverageGasPrice(ctx context.Context, epoch eth.BlockID) (*big.Int, error) {
+	_, txs, err := ba.l1.InfoAndTxsByHash(ctx, epoch.Hash)
+	if err != nil {
+		return nil, err
+	}
+	count := 0
+	var sum big.Int
+	for _, tx := range txs {
+		if tx.GasPrice().Cmp(common.Big0) <= 0 {
+			continue
+		}
+		sum.Add(&sum, tx.GasPrice())
+		count += 1
+	}
+	if count == 0 {
+		return bscDefaultGasPrice, nil
+	}
+	return sum.Div(&sum, big.NewInt(int64(count))), nil
+}
+
+type gasPriceWrapper struct {
+	eth.BlockInfo
+	gasprice *big.Int
+}
+
+var _ (eth.BlockInfo) = (*gasPriceWrapper)(nil)
```

Comment:
What is the purpose of this line?

Reply:
check the gasPriceWrapper implement eth.BlockInfo at compile time.